pages/blog/[article].js

```diff
@@ -0,0 +1,44 @@
+import Content from 'components/Content/Content';
+import Head from 'components/head';
+import HeroBanner from 'components/HeroBanner/HeroBanner';
+import PropTypes from 'prop-types';
+import { getArticlePath } from 'common/utils/path-utils';
+import ErrorPage from 'next/error';
+
+BlogArticle.propTypes = {
+  Article: PropTypes.object,
+  statusCode: PropTypes.number,
+};
+
+BlogArticle.defaultProps = {
+  Article: {},
+  statusCode: null,
+};
+
+BlogArticle.getInitialProps = async ({ query }) => {
+  const articlePath = getArticlePath(`${query.article}.mdx`);
+  const props = {};
+  if (articlePath === '') {
+    props.statusCode = 404;
+  } else {
+    // eslint-disable-next-line prefer-template
+    const Article = await import('../../blogArticles' + articlePath);
+    props.Article = Article;
+  }
+  return props;
+};
+
+function BlogArticle({ Article, statusCode }) {
+  if (statusCode) {
+    return <ErrorPage statusCode={statusCode} />;
+  }
```

Comment:
Assuming you get rid of the path-checking function and use `pathname` instead of `query`, we won't need to deal with manually handling 404 - Next.js will throw correctly automatically 🙏

Reply:
yep. after implementing ``getStaticProps``, Next.js handles 404 👍